assets/javascripts/lib/apply-template.js

```diff
@@ -47,16 +49,16 @@ export function prepareTemplate(templateTitle, templateContent, model) {
 }
 
 export function insertTemplateIntoComposer(
-  context,
+  model,
   { templateTitle, templateContent }
 ) {
-  const model = context.get("model");
-
   // insert the title if blank
   if (model && !model.title) {
     model.set("title", templateTitle);
   }
 
   // insert the content of the template in the compose
-  context.appEvents.trigger("composer:insert-block", templateContent);
```

Comment:
The previous pattern of passing `appEvents` in was better. `getOwner` in this case relies on our hack that falls back to basically a global variable when used with an object that doesn't have an owner (`this` in a standalone function)

Reply:
I'd like to skip this change.

I have a follow up PR that extract some logic into a service and will change this line:

https://github.com/discourse/discourse-templates/compare/templates-2023...dev/templates-service